File: src/features/inspection/service/inspectionService.js
import { createCrudService } from '../../../shared/service/createCrudService';
import { createInspectionCategory } from '../domain/InspectionCategory';
import { inspectionRepository } from '../infrastructure/inspectionRepository';

export const inspectionService = createCrudService({
  repository: inspectionRepository,
  createEntity: createInspectionCategory,
  idField: 'inspectionCategoryId',
});

File: backend/src/main/java/com/oose/labsafety/inspection/service/InspectionService.java
package com.oose.labsafety.inspection.service;

import com.oose.labsafety.common.service.CrudService;
import com.oose.labsafety.inspection.domain.InspectionCategory;
import com.oose.labsafety.inspection.infrastructure.InspectionRepository;
import org.springframework.stereotype.Service;

@Service
public class InspectionService extends CrudService<InspectionCategory> {
    public InspectionService(InspectionRepository repository) {
        super(repository);
    }
}
